### PR TITLE
feat(headless): add method for first search after standalone redirect

### PR DESCRIPTION
**Pull request by samisayegh**
https://coveord.atlassian.net/browse/KIT-870

### CODE REVIEW

File: packages/headless/src/app/search-engine/search-engine.ts [hunk 1]
```diff
@@ -52,6 +57,15 @@ export interface SearchEngine<State extends object = {}>
...
   executeFirstSearch(analyticsEvent?: SearchAction): void;
+
+  /**
+   * Executes the first search, and logs the analytics event that triggered a redirection from a standalone search box.
+   *
+   * @param analytics - The standalone search box analytics data.
+   */
+  executeFirstSearchAfterStandaloneSearchBoxRedirect(
+    analytics: StandaloneSearchBoxAnalytics
+  ): void;
```
Comment (samisayegh) [inline packages/headless/src/app/search-engine/search-engine.ts:68]: I took into account [this comment](https://github.com/coveo/ui-kit/pull/1042/files#r681740085), but felt keeping an explicit name would reduce confusion on when it should be used. I feel the length is _okay_, but I am open to other suggestions,

File: packages/samples/headless-react/src/pages/SearchPage.tsx [hunk 2]
```diff
@@ -377,20 +377,10 @@ export class SearchPage extends Component {
...
-    );
-
-    const event =
-      metadata && cause === 'omniboxFromLink'
-        ? logOmniboxFromLink(metadata)
-        : logSearchFromLink();
 
     this.engine.dispatch(updateQuery({q: value}));
-    this.engine.executeFirstSearch(event);
+    this.engine.executeFirstSearchAfterStandaloneSearchBoxRedirect(analytics);
```
Comment (samisayegh) [inline packages/samples/headless-react/src/pages/SearchPage.tsx:383]: fyi @jpmarceau, I'm adding this method to make it easier to set up a standalone search box,

File: packages/atomic/package.json [hunk 3]
```diff
@@ -27,7 +27,7 @@
     "build:locales": "node --max-old-space-size=4096 ./scripts/split-locales.js",
     "build:stencil": "node --max_old_space_size=4096 ./node_modules/@stencil/core/bin/stencil build",
     "start": "npm run build:locales && npm run build:stencil -- --dev --watch --serve",
-    "start:prod": "npx serve www -l 3333",
+    "start:prod": "npx -y serve www -l 3333",
```
Comment (ThibodeauJF) [inline packages/atomic/package.json:30]: what does that do?

File: packages/headless/src/app/search-engine/search-engine.ts [hunk 4]
```diff
@@ -24,11 +24,16 @@ import {
...
+import {
+  logInterfaceLoad,
+  logOmniboxFromLink,
+  logSearchFromLink,
+} from '../../features/analytics/analytics-actions';
 import {firstSearchExecutedSelector} from '../../features/search/search-selectors';
 import {SearchAppState} from '../../state/search-app-state';
 import {SearchThunkExtraArguments} from '../search-thunk-extra-arguments';
 import {SearchAction} from '../../features/analytics/analytics-utils';
+import {StandaloneSearchBoxAnalytics} from '../../controllers';
```
Comment (ThibodeauJF) [inline packages/headless/src/app/search-engine/search-engine.ts:36]: possible circular dependency here? 🤔

**Review by ThibodeauJF — APPROVED** [review]

**Review by nbuissoncoveo — APPROVED** [review]

File: packages/atomic/package.json (hunk 3 above)
Comment (samisayegh) [reply packages/atomic/package.json:30]: The `-y` skips the manual approval to install the server,

**Pull request merged**